src/plugins/typechecker/Elektra/Typechecker.hs.in

```diff
@@ -119,12 +117,15 @@ keyInterpreter specFile = do
     isFun (Fun _) = True
     isFun _ = False
 
-errorString :: InterpreterError -> String
-errorString (WontCompile es) = intercalate "\n" (header : map unbox es)
+errorString :: String -> InterpreterError -> String
+errorString f (WontCompile es) = intercalate "\n" (header : filterErrors es)
   where
-    header = "ERROR: Won't compile:"
+    header = "Typechecking or compilation failed:"
```

Comment:
Can you separate type-checking and compilation errors?

Reply:
Usually typechecking errors mean unresolveable constraints indicated by "Could not deduce". 
Maybe not the most reliable method as i can only check the error message at this point and have no further information about it, but i guess it should be fine enough.